token/src/token.rs

```diff
@@ -0,0 +1,190 @@
+    issuance_predicate: Predicate,
+    metadata: Vec<u8>,
+}
+
+impl Token {
+    /// Constructs a new Token.
+    pub fn new(pred: Predicate, metadata: Vec<u8>) -> Self {
+        Token {
+            issuance_predicate: pred,
+            metadata: metadata,
+        }
+    }
+
+    /// Returns program that issues specified quantity of Token.
+    pub fn issue<'a>(program: &'a mut Program, token: &Token, qty: u64) -> &'a mut Program {
+        program
+            .push(Commitment::blinded(qty)) // stack: qty
+            .var() // stack: qty-var
+            .push(Commitment::unblinded(Value::issue_flavor(
+                &token.issuance_predicate,
+                Data::Opaque(token.metadata.clone()),
+            ))) // stack: qty-var, flv
+            .var() // stack: qty-var, flv-var
+            .push(Data::Opaque(token.metadata.to_vec())) // stack: qty-var, flv-var, data
+            .push(token.issuance_predicate.clone()) // stack: qty-var, flv-var, data, flv-pred
+            .issue() // stack: issue-contract
+            .sign_tx() // stack: issued-value
+    }
+
+    /// Returns program that issues specified quantity of Token,
+    /// outputting it to the destination Predicate.
+    pub fn issue_to<'a>(
+        program: &'a mut Program,
+        token: &Token,
+        qty: u64,
+        dest: &Predicate,
+    ) -> &'a mut Program {
+        Token::issue(program, token, qty)
+            .push(dest.clone())
+            .output(1)
+    }
+
+    /// Returns program that retires a Token.
+    /// TBD: better symmetry here to accept a Token object
+    pub fn retire<'a>(
+        program: &'a mut Program,
+        qty: CompressedRistretto,
+        flv: CompressedRistretto,
+        pred: &Predicate,
+        txid: TxID,
+    ) -> &'a mut Program {
+        let output = (Commitment::Closed(qty), Commitment::Closed(flv));
+        program
+            .push(Data::Input(Box::new(Input::new(
+                vec![output],
+                pred.clone(),
+                txid,
+            ))))
+            .input()
+            .sign_tx()
+            .retire()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn issue_to() {
+        let (tx, _, txlog) = {
+            let issue_key = Scalar::from(1u64);
+            let dest_key = Scalar::from(2u64);
+            let nonce_key = Scalar::from(3u64);
+            let usd = Token::new(
+                Predicate::Key(VerificationKey::from_secret(&issue_key)),
+                b"USD".to_vec(),
+            );
+            let dest = Predicate::Key(VerificationKey::from_secret(&dest_key));
+            let program = Program::build(|p| {
+                Token::issue_to(p, &usd, 10u64, &dest)
+                    .push(Predicate::Key(VerificationKey::from_secret(&nonce_key)))
+                    .nonce()
+                    .sign_tx()
+            })
+            .to_vec();
+            match build(program, &usd, vec![issue_key, nonce_key]) {
```

Comment:
you can just `build().unwrap()` - it will return a good transaction or panic in place just like `assert!`. So no need for `match` statement.